capstone/fitness/src/main/java/com/google/sps/servlets/AbstractAuthServlet.java

```diff
@@ -1,42 +1,150 @@
-    response.setContentType("text/html;");
-    response.getWriter().println(nickname + " has authorized this app.");
-  }
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
+    response.setContentType("application/json;");
+    
+    // Build calendar. 
+    String userId = getUserId(request);
+    Credential credential = Utils.newFlow().loadCredential(userId);
+    calendar = new Calendar.Builder(
+    new UrlFetchTransport(),
+    new JacksonFactory(),
+    credential).setApplicationName(APPLICATION_NAME).build();
 
+    // Might be an extra step. 
+    com.google.api.services.calendar.Calendar.CalendarList.List listRequest = calendar.calendarList().list();
+    listRequest.setFields("items(id)").setMaxResults(1);
+    CalendarList feed = listRequest.execute();
+   
+    ArrayList<String> result = new ArrayList<String>();
+      if (feed.getItems() != null) {
+        for (CalendarListEntry entry : feed.getItems()) {
+          result.add(entry.getId());
+        }}
+    // TODO (@piercedw) : Store this result in datastore.
+    String id = result.get(0);
+    String jsonId = gson.toJson(result.get(0));
+    response.getWriter().println((jsonId));    
+ 
+    scheduler = new Scheduler(exerciseDuration);
+    
+    String wks = (DataHandler.getData("weeksToTrain",DataHandler.getUser()));
+    int weeksToTrain = Integer.parseInt(wks);
+    
+    int daysAvailable = weeksToTrain * Time.weeksToDays;
+
+    String goalSteps = DataHandler.getGoalSteps();
+
+    ArrayList<JsonGoalStep> goalStepsArray = gson.fromJson(goalSteps, new TypeToken<List<JsonGoalStep>>(){}.getType());
+    List<String> exercises = new ArrayList<String>();
+    for(JsonGoalStep goal: goalStepsArray){
+        exercises.add(goal.getName());
+    }
+    
+    int timesPerWeek = daysAvailable/ exercises.size(); 
+
+    // // Sets minSpan to 7:00 AM the next day, and maxSpan to 7PM the next day.
+    LocalDateTime now = LocalDateTime.now();  
+    DateTimeFormatter myDtf = DateTimeFormatter.ofPattern("YYYY-MM-dd");  
+    String nextDayOfMonth = myDtf.format(now.plusDays(1));
+    String nextDayStartTime = "T11:00:00+00:00";
+    String nextDayEndTime = "T23:00:00+00:00";
+    String minRCF3339 = nextDayOfMonth + nextDayStartTime;
+    String maxRCF3399 = nextDayOfMonth + nextDayEndTime;
+    DateTime minSpan = new DateTime(minRCF3339);
+    DateTime maxSpan = new DateTime(maxRCF3399);
+
+    // Use the scheduler to schedule one exercise per day. 
+    int y = 0;
+    for (int x = 0; x<daysAvailable; x+=timesPerWeek){
+        if (y>=exercises.size()){
+          break;
+        }
+        else{
+          List<Event> currentlyScheduledEvents = this.getEventsInTimespan(minSpan, maxSpan);
+          Event exerciseEvent = scheduler.getFreeTime(minSpan, maxSpan, currentlyScheduledEvents);
+          exerciseEvent.setSummary("Marathon App: " + exercises.get(y));
+          exerciseEvent.setColorId("4");
```

Comment:
Just curious, is there a specific reason you chose this color id? I would recommend pulling out the value to a variable as well.

Reply:
Yeah. I can make it a variable. I think I had originally just made it that color ID so that it was easier to see on my calendar when I did the demo, since otherwise all of the events are the default color and you can't really tell them apart. I figure that if we end up creating a color scheme for the app I can set the color to something that matches that later on.